src/const_choice.rs

```diff
@@ -164,11 +174,144 @@ impl PartialEq for ConstChoice {
     }
 }
 
+/// An equivalent of `subtle::CtOption` usable in a `const fn` context.
+#[derive(Debug, Clone)]
+pub struct ConstOption<T> {
```

Comment:
Perhaps this should be `ConstCtOption` so the mapping is:

- `subtle::Choice` => `ConstChoice`
- `subtle::CtOption` => `ConstCtOption`

?

Reply:
Changed the name.

> I also find ConstOption a little confusing as Option is already const-friendly

It is actually not, only `is_some()`, `is_none()` and `as_ref()` are `const`.

Reply:
Well, pattern matching gets you most of the functionality. The rest of that is just sugar.